### PR TITLE
feat(repo-cli): document the detached until-ready recipe and observe merge-ready rows

### PR DESCRIPTION
**Pull request by kriegcloud**
## What this PR delivers (ttc B7, PR2 of 2)

Ruling 48's deferred half, now that B5 (#1143) and B7 PR1 (#1149) are both on main:

- `yeet inbox ack <id> --observed` accepts `pr-merge-ready` rows (ruling 46) alongside
  `proof-job-finished`; the admitting kinds are one `LiteralKit` and the refusal names both.
- `yeet inbox list` treats merge-ready rows as live rather than wave-scoped: the merge loop
  supersedes them itself with a `fix-sha` receipt on the next push, so the remediation wave
  never owns them (the row rendered `P1 unknown` before).
- The yeet skill's step 6 and the AGENTS.md closeout bullet name the detached recipe first:
  `yeet monitor --until-ready --detach`, then `yeet job wait <jobId>` from a background tool
  call (0 green / 1 red / 2 terminated), with attached `--until-ready` as the fallback when the
  user manager is unreachable. A wiring test proves `--until-ready --detach` reaches the
  detached submit before route legality.
- The idle-wake spike (`FileChanged` + `asyncRewake`) is written up in
  `goals/time-to-certainty/research/b7-implementation.md` as proposed and unverified: the hook
  JSON an operator would try in `.claude/settings.local.json`, the observable that proves a
  wake, and the risks. No tracked hook wiring changes.
- The scratchpad PR watcher is retired with a receipt in `research/OPPORTUNITIES.md`
  (the two measured runs on #1149 are the evidence); PLAN.md B7 is checked.
- **Merged `main` after B8 (#1155):** this branch's fix for the empty-rollup false `settle-timeout`
  (seen on #1159) is dropped. B8's amendment to rulings 54–55 landed a stronger one first: both
  loops remember every registered context and keep an absent one `pending`, and a conflicting
  head reads `base-conflict`. Rulings 50–57 stay B8's. The skill and AGENTS.md now also carry
  main's `ready-for-heavy` label rule and note that a reboot needs the detached job re-submitted.
  `proof-job.test.ts` asserts the widened `--observed` refusal. Receipts for the reboot, the
  duplicate fix, and the coverage-ratchet evidence from the closed #1159 are in `OPPORTUNITIES.md`.
- **Review round 3:** `Inbox.ts` exports `YeetInboxObservedRowKind` and the type guard
  `yeetInboxRowIsObserved`, which both `--observed` admission and the inbox list's always-live
  rule now read; `Ack.ts` documents both observed row kinds; the idle-wake hook example tests the
  captured `jq` result instead of the hook's stdin.

The Codex lane for this PR died on the OpenAI OAuth quota before its first edit (receipt in
`OPPORTUNITIES.md`), so the orchestrator implemented the bounded items directly.

## Verification

| Gate (on `2ab8652608`: main through #1164 plus review round 3) | Result |
| --- | --- |
| `bunx vitest run` (inbox, inbox-porcelain, inbox-view, command-wiring, proof-job, settle, monitor-ready, watch-mode) Node | 8 files, 285 tests pass |
| `bunx --bun vitest run` same suites | 8 files, 285 tests pass |
| `bunx turbo run check --filter=@beep/repo-cli` | pass |
| `bun run beep quality test-tsgo` | pass |
| `bun run docgen:local` | pass |
| `bun run beep quality fallow audit --check --base origin/main` / `fallow health --check` | pass |
| `bun run lint:schema-first` / `bun run lint:circular` / `bun run beep quality knip` | pass |
| oxlint + ESLint (`--max-warnings=0`) on touched sources | clean |
| `bun run beep lint` (full) | exit 0 |
| Idle-wake example, command extracted from the doc and run on one-row inbox fixtures | merge-ready and P0 rows exit 2 with the message on stderr; P1 and malformed rows exit 0 |

## Dogfood

This PR is babysat with the recipe it documents (`yeet monitor --until-ready --detach` + `yeet job wait`); transcript and push→ready wall clock land in a follow-up comment once the loop reports `merge-ready: yes`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by kriegcloud** [comment]
## Confidence Score: 4/5

Sound B7 PR2 (observed merge-ready + always-live list + detach recipe docs); one leftover PLAN placeholder to fix.

## Findings

1. **P2 · Replace PLAN.md PR2 placeholder with #1161** → `goals/time-to-certainty/PLAN.md`

<details>
<summary>Fix with agent prompt</summary>

### Issue 1
goals/time-to-certainty/PLAN.md:82
Replace the literal placeholder `PR2 <PR2>:` with `PR2 #1161:` so the B7 checkbox cites this PR the same way the adjacent `PR1 #1149` citation does.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
</details>

<details open>
<summary><h3>Summary</h3></summary>

- Documents and wires ruling 48's deferred half: `yeet inbox ack --observed` admits `pr-merge-ready` via a closed `LiteralKit`, `yeet inbox list` treats merge-ready as always-live (wave never owns those rows), and the yeet skill / AGENTS.md name `monitor --until-ready --detach` + `job wait` as the canonical babysit recipe (attached fallback when the user manager is down). Idle-wake spike stays proposed/unverified; inventory regen is noise.
- Verified critical angles: ObservedRowKind admits only the two informational kinds (check-failed still refused); always-live merge-ready matches MonitorLoop's fix-sha ack of the prior `announcedRow` on head change; LiteralKit import matches repo pattern; wiring test intentionally only proves detach-submit reachability via the `--detach`/`--plan` conflict (same pattern as other detach smokes); detach path itself landed in B7 PR1 (#1149).
- Remaining: P2 PLAN placeholder (above). Adjacent note (not in diff): `Ack.ts` still describes `YeetAckObservedResolution` as proof-job-only.
- Risk tier **L2** · seats: Domain Modeler, Reliability Adversary, Junior Engineer, Category Theorist · verify-then-triage SoD.
- Graft: DankStation offline — `deep_meaning_used: false`; no graft CLI/index on box; review from PR head checkout `6d60edb9` + GitHub MCP file patches.
- mergeable_state **blocked** — COMMENT only, do not APPROVE.

</details>

<details open>
<summary><h3>Diagram</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
  Ready[until-ready settle] --> Announce[append pr-merge-ready P1]
  Announce --> List["inbox list: always live"]
  Announce --> AckObs["ack --observed via LiteralKit"]
  Push[head change] --> FixSha["MonitorLoop fix-sha ack prior announcedRow"]
  Recipe["skill/AGENTS: monitor --until-ready --detach"] --> Wait["job wait 0/1/2"]
```

</details>

<sub>Reviews (1) · Last reviewed commit: ["docs(ttc): record the early-push ref-lock race receipt"](https://github.com/beep-effect/beep-effect/commit/6d60edb9c39f8885b038fbf65d04c608b70531bf)</sub>

File: goals/time-to-certainty/PLAN.md [hunk 1]
```diff
@@ -74,11 +74,13 @@ orchestrator owns schemas, contracts, and judgment.
...
       P1 otherwise) acknowledged by the new `observed` resolution; `yeet job list|status|wait|logs|cancel`.
       Brief and evidence: `research/b5-brief.md`, `research/b5-implementation.md`.
-- [~] B7 `yeet monitor --until-ready` as the canonical PR babysit path (rulings 41–49): settle on
+- [x] B7 `yeet monitor --until-ready` as the canonical PR babysit path (rulings 41–49): settle on
       the base ruleset's expected contexts with tolerated matrix parents and `--settle-timeout`,
       automatic read-first closeout on settle, an exit-0 `ready` terminal, required-only exit
       codes in every monitor mode, one P1 `pr-merge-ready` inbox row per head, push→ready
-      measurement. PR1 carries the attached loop (this PR); PR2 after B5: detach recipe + observed ack.
+      measurement. PR1 #1149 merged 2026-09-16 (push→ready 1h 0m 7s on its own babysit, exit 0 with
+      two optional reds); PR2 <PR2>: detach recipe, `--observed` for merge-ready rows, inbox-list
```
Comment (kriegcloud) [inline goals/time-to-certainty/PLAN.md]: **P2 · Replace PLAN.md PR2 placeholder with #1161**

B7 is marked done with a literal `PR2 <PR2>:` stub beside the concrete `PR1 #1149` citation. Readers (and later greps) will miss that this work shipped as #1161.

<details>
<summary>Prompt To Fix With AI</summary>

```markdown
This is a comment left during a code review.
Path: goals/time-to-certainty/PLAN.md
Line: 82

Comment:
**Replace PLAN.md PR2 placeholder with #1161**

Replace `PR2 <PR2>:` with `PR2 #1161:` so the checked B7 bullet cites this PR consistently with `PR1 #1149`.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
```

</details>

**Review by kriegcloud — COMMENTED** [review]
L2 Greptile-format review: one P2 PLAN placeholder; ObservedRowKind + merge-ready liveness + detach recipe docs look sound. mergeable_state blocked — COMMENT only.

**Comment by kriegcloud** [comment]
> Supersedes prior greptile_style_summary comment 5695734357 (GitHub MCP cannot PATCH issue comments).

## Confidence Score: 5/5

Production ready on the Ruling 50 delta; prior PLAN.md P2 is fixed; no new actionable findings (mergeable_state blocked — COMMENT only).

<details>
<summary>Fix with agent prompt</summary>

No new actionable issues from this re-review.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
</details>

<details open>
<summary><h3>Summary</h3></summary>

- **What:** Re-review of B7 PR2 after Ruling 50 (`6bb71a24`) + docs tip (`4af0f48c`): a head that has registered never re-enters the settle registration window; empty `gh pr checks` after registration is a suspect census (`YeetCensusRead` / `yeetCensusReadIsSuspect`), counted against the five-poll error budget in the merge loop and ridden through on the last verdict in watch mode. `deriveSettleVerdict` stays pure/unchanged.
- **Prior P2 FIXED:** `goals/time-to-certainty/PLAN.md` now cites `PR2 #1161:` (was `PR2 <PR2>:`). Thread `PRRT_kwDOPbO_N86i4Q2z` / `discussion_r4024820484` resolved.
- **Latch:** `MonitorHeadState.registered` / `WatchSettleState.registered` reset to `false` on new head (`observeMonitorHead`, WatchSettleState recreation on `headSha` change); monotone OR within a head.
- **Remaining:** No new actionable issues.
- **Risk:** L2 · seats Domain Modeler, Reliability Adversary, Junior Engineer, Category Theorist · `deep_meaning_used: false` (DankStation offline) · `mergeable_state: blocked` — do not APPROVE.
- **Blast:** Local to yeet settle/monitor/watch; inventory regen noise ignored.

</details>

<details open>
<summary><h3>Diagram</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
  Poll[Poll checks census] --> Reg{registered?}
  Reg -->|false + empty| Win[Registration window<br/>deriveSettleVerdict]
  Reg -->|false + non-empty| Latch[registered := true]
  Latch --> Derive[deriveSettleVerdict]
  Reg -->|true + non-empty| Derive
  Reg -->|true + empty| Suspect[yeetCensusReadIsSuspect]
  Suspect --> Merge[Merge loop: poll-error budget 1..5]
  Suspect --> Watch[Watch: keep last verdict]
  HeadChg[New headSha] --> Reset[registered := false]
```

</details>

<sub>Reviews (2) · Last reviewed commit: ["docs(repo-cli): tag the census predicate and name PR2 in the B7 plan line"](https://github.com/beep-effect/beep-effect/commit/4af0f48c931b37d66900ace85562b3efaac5a6d0)</sub>

**Review by kriegcloud — COMMENTED** [review]
L2 Greptile-format re-review (r2) at 4af0f48c: Ruling 50 latch verified; prior PLAN P2 FIXED; no new inline findings. Confidence 5/5. mergeable_state blocked — COMMENT only.

File: packages/tooling/tool/cli/src/commands/Yeet/internal/WatchMode.ts [hunk 2]
```diff
@@ -762,22 +769,37 @@ export const runYeetWatchStream = Effect.fn("Yeet.runYeetWatchStream")(function*
...
         })
       );
     }
     const currentHead = O.getOrThrow(head);
+    const checks = A.map(snapshot.checks, (check) =>
+      YeetSettleCheck.make({ name: check.name, outcome: check.outcome, required: check.required })
+    );
+    // Ruling 50: an empty census after registration is a bad read; keep the
+    // last verdict instead of re-opening the registration budget.
+    if (yeetCensusReadIsSuspect(YeetCensusRead.make({ registered: currentHead.registered, checks }))) {
```
Comment (kriegcloud) [inline packages/tooling/tool/cli/src/commands/Yeet/internal/WatchMode.ts]: **P2 · Stop registration-wait stderr after head already registered**

This suspect branch correctly keeps the last verdict and logs `YEET_CENSUS_SUSPECT_MESSAGE`, but `reportWatchRegistrationWait` (~L661) still keys only on empty `snapshot.checks`, so the next tick logs "no checks registered for head … yet (N/PATIENCE)" and `emptyPolls` keeps climbing. Gate that narration (and preferably patience accounting) on `!WatchSettleState.registered`.

<details>
<summary>Prompt To Fix With AI</summary>

```markdown
This is a comment left during a code review.
Path: packages/tooling/tool/cli/src/commands/Yeet/internal/WatchMode.ts
Line: 783-785

Comment:
**Stop registration-wait stderr after head already registered**

Only emit the registration-patience stderr while `WatchSettleState.registered` is false. After a suspect empty census, do not claim checks have not registered yet.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
```

</details>

File: packages/tooling/tool/cli/src/commands/Yeet/internal/MonitorLoop.ts [hunk 3]
```diff
@@ -1190,13 +1196,35 @@ const reportMonitorSettleTransition = Effect.fn("YeetMonitorLoop.reportSettleTra
...
+};
+
 const settleAndCloseoutMonitorHead = Effect.fn("YeetMonitorLoop.settleAndCloseout")(function* (
   context: RepoRunContext,
   options: YeetMonitorUntilMergedOptions,
   observation: MonitorObservation
 ) {
   if (O.isNone(observation.snapshot.remote.headSha)) return Result.succeed(observation);
-  const current = O.getOrThrow(observation.poll.head);
+  const registered = registerMonitorCensus(observation.poll, observation.snapshot.remote.checks);
```
Comment (kriegcloud) [inline packages/tooling/tool/cli/src/commands/Yeet/internal/MonitorLoop.ts]: **P1 · Gate closeout reread through yeetCensusReadIsSuspect**

`registerMonitorCensus` runs here before the first settle, but after a successful `closeoutMonitorHead` reread the loop calls `settleMonitorHead(next, current, policy)` (~L1245) with no second suspect check. An empty successful reread on an already-`registered` head re-opens `deriveSettleVerdict`'s registration window; with a large `waitedMs` that becomes `settle-timeout` — the same #1159 class this gate closed on the ordinary poll path.

<details>
<summary>Prompt To Fix With AI</summary>

```markdown
This is a comment left during a code review.
Path: packages/tooling/tool/cli/src/commands/Yeet/internal/MonitorLoop.ts
Line: 1225-1245

Comment:
**Gate closeout reread through yeetCensusReadIsSuspect**

After `closeoutMonitorHead` succeeds, run the reread census through `registerMonitorCensus` / `yeetCensusReadIsSuspect` before `settleMonitorHead`. On suspect, fail the poll against the five-poll error budget and do not derive registration/settle-timeout. Add a TestClock test for empty post-closeout reread.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
```

</details>

**Review by kriegcloud — COMMENTED** [review]
L2 Greptile-format re-review (Reviews 2): ruling 50 main path looks sound; P1 closeout-reread still bypasses suspect census gate; P2 watch registration narration after registered. Prior PLAN P2 resolved. mergeable_state blocked — COMMENT only.

**Comment by kriegcloud** [comment]
## Confidence Score: 5/5

Tip looks production-ready on the merge delta: PLAN keeps B7 PR2 #1161 and takes main's closed B8; owned Yeet code unchanged vs r4; no new actionable findings.

<details open>
<summary><h3>Summary</h3></summary>

- **What changed at this tip (`2ab86526`)**: merge of `origin/main` picking up #1152 (local Codex Security scan ingest) and #1164 (B9 capture; B8 closed). Conflict resolution: `PLAN.md` keeps this branch's B7 `PR2 #1161:` line and takes main's checked B8 (with #1164/#1165 skip-lane evidence); `OPPORTUNITIES.md` keeps both sides' receipts. Main-only Codex/explorations paths are not part of this PR's diff vs `main`.
- **Owned code vs r4 (`2664cff6`)**: unchanged — `YeetInboxObservedRowKind` / `yeetInboxRowIsObserved` still shared by `--observed` admission and always-live liveness; Ack observed docs and idle-wake capture+exit-2 example untouched.
- **Prior finding dispositions** (all threads still resolved):
  - [discussion_r4024820484](https://github.com/beep-effect/beep-effect/pull/1161#discussion_r4024820484) P2 PLAN `<PR2>` → **FIXED** (still `#1161` at tip).
  - [discussion_r4027135576](https://github.com/beep-effect/beep-effect/pull/1161#discussion_r4027135576) / [discussion_r4025123741](https://github.com/beep-effect/beep-effect/pull/1161#discussion_r4025123741) idle-wake → **FIXED**.
  - [discussion_r4027135585](https://github.com/beep-effect/beep-effect/pull/1161#discussion_r4027135585) ObservedRowKind kit → **FIXED**.
  - [discussion_r4025044871](https://github.com/beep-effect/beep-effect/pull/1161#discussion_r4025044871) / [discussion_r4025123759](https://github.com/beep-effect/beep-effect/pull/1161#discussion_r4025123759) closeout reread → **FIXED/OBSOLETE by B8**.
  - [discussion_r4025044867](https://github.com/beep-effect/beep-effect/pull/1161#discussion_r4025044867) WatchMode registration-wait stderr → **OBSOLETE for this PR**.
- **Remaining**: No new actionable P0/P1/P2 on this PR's owned files.
- **Risk tier**: L0 · seats: Junior Engineer, Domain Modeler, Hermeneut · `deep_meaning_used: false` (DankStation offline; no graft deep refresh).
- `mergeable_state: blocked` (Vercel rate-limit reds) → COMMENT only — do not APPROVE.

</details>

<details open>
<summary><h3>Diagram</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
  Main["main + #1152 + #1164"] --> Tip[2ab86526 tip]
  Tip --> Plan["PLAN: keep B7 #1161 / take B8 closed"]
  Tip --> Opp[OPPORTUNITIES both receipts]
  Tip --> Owned["Yeet inbox/Ack/idle-wake unchanged"]
  Plan --> Clean[no NEW findings]
  Owned --> Clean
```

</details>

<sub>Reviews (6) · Last reviewed commit: ["Merge remote-tracking branch 'origin/main' into ttc/b7-pr2-detach-recipe"](https://github.com/beep-effect/beep-effect/commit/2ab865260817166a08f4fd832403f91ebfb53648)</sub>

File: goals/time-to-certainty/research/b7-implementation.md [hunk 4]
```diff
@@ -609,3 +609,64 @@ is the shared pure rule for the watch. Two TestClock cases pin the dogfood shape
...
+{
+  "hooks": {
+    "FileChanged": [
+      {
+        "matcher": ".beep/inbox/failures.ndjson",
+        "hooks": [
+          {
+            "type": "command",
+            "asyncRewake": true,
+            "command": "tail -n 1 .beep/inbox/failures.ndjson | jq -r 'select(.kind == \"pr-merge-ready\" or .severity == \"P0\") | \"[yeet] inbox: \" + .kind + \" \" + .id' 1>&2; test -s /dev/stdin || exit 0; exit 2"
```
Comment (kriegcloud) [inline goals/time-to-certainty/research/b7-implementation.md]: **P2 · Fix idle-wake jq/stdin example**

The pipeline still sends filtered `jq` output to stderr (`1>&2`), then runs `test -s /dev/stdin`, which reads the hook's inherited stdin — not the filter result. Exit code therefore does not depend on whether the inbox row matched. Capture the filtered message, test that value, emit it to stderr, and exit 2 only when non-empty (proposed/unverified research, but operators may copy this recipe into `.claude/settings.local.json`).

<details>
<summary>Prompt To Fix With AI</summary>

```markdown
This is a comment left during a code review.
Path: goals/time-to-certainty/research/b7-implementation.md
Line: 657

Comment:
**Fix idle-wake jq/stdin example**

Update the idle-wake command example so the filtered jq result is captured in a variable, its non-empty state is tested directly, and the captured message is emitted to stderr; exit 2 only when a matching inbox row exists, otherwise exit 0. Preserve the existing filter and message format.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
```

</details>

File: packages/tooling/tool/cli/src/commands/Yeet/internal/InboxPorcelain.ts [hunk 5]
```diff
@@ -55,6 +56,12 @@ const $I = $RepoCliId.create("commands/Yeet/internal/InboxPorcelain");
 
 const isoNow = DateTime.now.pipe(Effect.map(DateTime.formatIso));
 
+// Informational rows acknowledged by observation (ruling 37 for proof jobs, ruling 46 for
+// merge-ready announcements); every other kind needs an attributed closing move.
+const ObservedRowKind = LiteralKit(["proof-job-finished", "pr-merge-ready"]);
+const isObservedRowKind = S.is(ObservedRowKind);
+const yeetInboxRowAcceptsObserved = (row: YeetInboxRow): boolean => isObservedRowKind(row.kind);
```
Comment (kriegcloud) [inline packages/tooling/tool/cli/src/commands/Yeet/internal/InboxPorcelain.ts]: **P2 · Export ObservedRowKind for InboxView always-live subset**

`ObservedRowKind` LiteralKit admits `proof-job-finished` | `pr-merge-ready` for `--observed`, while `yeetInboxRowLiveness` re-lists the same two tags with raw `===` beside `sibling-collision` / `local-shard-failed`. Adding a third observed kind can desync ack admission vs list liveness. Export one kit (or shared predicate) for the observed-admissible tags and reuse it from both call sites; keep the broader always-live set intact.

<details>
<summary>Prompt To Fix With AI</summary>

```markdown
This is a comment left during a code review.
Path: packages/tooling/tool/cli/src/commands/Yeet/internal/InboxPorcelain.ts
Line: 61-63

Comment:
**Export ObservedRowKind for InboxView always-live subset**

Export the observed-admissible LiteralKit (or a shared predicate) and use it from both `--observed` admission and the always-live branch in InboxView for those two kinds. Avoid duplicating the tags as raw string equality beside the kit.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
```

</details>

**Review by kriegcloud — COMMENTED** [review]
L1 re-review at 87204599 (B8 take-main). Prior P1 closeout FIXED/OBSOLETE by B8 `rememberRegistered`; prior WatchMode P2 obsolete for this PR (residual stderr on main noted in summary). Two new P2s on remaining docs/schema delta. COMMENT only — mergeable_state blocked.

File: packages/tooling/tool/cli/src/commands/Yeet/internal/WatchMode.ts (hunk 2 above)
Comment (kriegcloud) [reply packages/tooling/tool/cli/src/commands/Yeet/internal/WatchMode.ts]: **Disposition (r3 @ 87204599):** OBSOLETE for this PR — WatchMode took main via B8. `rememberRegistered` keeps settle pending across empty rollups, but `reportWatchRegistrationWait` still keys only on raw empty `snapshot.checks` (residual narration hole on main). Noted in summary; not re-filed on #1161.

File: packages/tooling/tool/cli/src/commands/Yeet/internal/MonitorLoop.ts (hunk 3 above)
Comment (kriegcloud) [reply packages/tooling/tool/cli/src/commands/Yeet/internal/MonitorLoop.ts]: **Disposition (r3 @ 87204599):** FIXED/OBSOLETE by B8 (#1155). This PR dropped its ruling-50 suspect flag when merging main. Post-closeout `settleMonitorHead` now feeds `rememberRegistered(current.registered, …).checks` into `deriveSettleVerdict`, so an empty reread after registration synthesizes pending rows instead of reopening registration/`settle-timeout`. Not re-filed on this PR.

File: goals/time-to-certainty/research/b7-implementation.md [hunk 6]
```diff
@@ -609,3 +609,63 @@ is the shared pure rule for the watch. Two TestClock cases pin the dogfood shape
...
```
[the remark this answers is not in the document]
Comment (kriegcloud) [reply goals/time-to-certainty/research/b7-implementation.md]: Fixed in 2664cff607. The example now captures the filtered `jq` result in `message`, prints it to stderr, and exits 2 only when it is non-empty; otherwise it exits 0. The old form tested the hook's inherited stdin, as you said. I extracted the JSON block from the doc and ran its command against one-row inbox fixtures: a `pr-merge-ready` row and a P0 `check-failed` row exit 2 with the `[yeet] inbox: <kind> <id>` line on stderr; a P1 `check-failed` row and a malformed line exit 0.

The outside-diff note on `Ack.ts` is fixed in the same commit: `YeetAckObservedResolution`, its annotation, and the `YeetAckResolution` details now name both `proof-job-finished` and `pr-merge-ready`, with no runtime change. The closeout-reread finding on `MonitorLoop.ts` is obsolete: this PR took `main`'s loop after B8 (#1155), whose `rememberRegistered` already keeps registered contexts pending on the post-closeout settle.

File: goals/time-to-certainty/research/b7-implementation.md (hunk 4 above)
Comment (kriegcloud) [reply goals/time-to-certainty/research/b7-implementation.md]: Fixed in 2664cff607. The example now captures the filtered `jq` result in `message`, prints it to stderr, and exits 2 only when it is non-empty; otherwise it exits 0. The old form tested the hook's inherited stdin, as you said. I extracted the JSON block from the doc and ran its command against one-row inbox fixtures: a `pr-merge-ready` row and a P0 `check-failed` row exit 2 with the `[yeet] inbox: <kind> <id>` line on stderr; a P1 `check-failed` row and a malformed line exit 0.

File: packages/tooling/tool/cli/src/commands/Yeet/internal/InboxPorcelain.ts (hunk 5 above)
Comment (kriegcloud) [reply packages/tooling/tool/cli/src/commands/Yeet/internal/InboxPorcelain.ts]: Fixed in 2664cff607. `Inbox.ts` now exports `YeetInboxObservedRowKind` (the LiteralKit) and `yeetInboxRowIsObserved`, a type guard narrowing to `Extract<YeetInboxRow, { kind: YeetInboxObservedRowKind }>`. `--observed` admission in `InboxPorcelain.ts` and the always-live branch of `yeetInboxRowLiveness` in `InboxView.ts` both call it; `sibling-collision` and `local-shard-failed` stay as the broader always-live checks. The guard keeps the union narrowing that the raw `===` chain gave the wave branch. A new test in `yeet-inbox-view.test.ts` pins the kit's two literals and checks that an observed row stays live under a moved wave while a check-failed row on the same wave is superseded.

**Review by kriegcloud — COMMENTED** [review]
L1 Greptile-format re-review (Reviews 5) at 2664cff6: both r3 P2s verified FIXED (shared YeetInboxObservedRowKind + idle-wake capture/exit-2); Ack docs aligned; no NEW inlines. Confidence 5/5. mergeable_state blocked — COMMENT only.

**Review by kriegcloud — COMMENTED** [review]
**L0 re-review @ `2ab86526`** — merge main (#1152 + #1164). PLAN keeps B7 PR2 #1161; owned Yeet code unchanged vs r4. Confidence **5/5**. P0:0 P1:0 P2:0 NEW. Priors remain FIXED/OBSOLETE. `mergeable_state: blocked` — COMMENT only, do not APPROVE. Full write-up in the greptile_style_summary comment.

**Comment by kriegcloud** [comment]
## Dogfood: babysat with the recipe this PR documents

This PR was babysat end to end with `bun run beep yeet monitor --until-ready --detach` and `bun run beep yeet job wait <jobId>`.

**Final job (`635e28e5`, submitted at head `2664cff607`, followed the push to `2ab8652608`):**

```text
[yeet] settle: required-pending
[yeet] settle: required-pending → base-conflict
[yeet] settle: registration
[yeet] settle: registration → required-pending
[yeet] closeout: 0 issue(s) for head 2ab8652
[yeet] settle: required-pending → settled
[yeet] merge-ready: yes; every hard criterion is green; awaiting the operator's merge; settle: settled; closeout bound
[yeet] merge-ready: yes; every hard criterion is green; awaiting the operator's merge; push→ready 28m 19s 464ms (pushed 2026-09-16T17:09:35.000Z, settled 2026-09-16T17:37:37.977Z, closeout 2026-09-16T17:37:43.803Z, ready 2026-09-16T17:37:54.464Z)
[yeet] merge-ready: yes; every hard criterion is green; hand the pull request to the operator
```

| Measure | Value |
| --- | --- |
| push→ready on the final head | 28m 19s |
| Loop exit | `merge-ready: yes`, process exit 0 |
| Remaining reds at ready | two Vercel deployments (daily rate limit), optional `Heavy / Coverage Regression` pending |

**What the run exposed:**

- **The base-conflict wait held.** `main` moved under the head (#1152, #1164). The loop reported `base-conflict`, stayed non-terminal, and followed the merge push without being re-submitted.
- **Two reboots stopped earlier jobs** (`a4300079`, `59deed9b`). A detached job survives a session restart but not a reboot; the skill and AGENTS.md now say to re-submit.
- **Bug: `yeet job wait` returned 2 (`terminated`) on this green loop.** The job record says `serviceResult: success`, `exitStatus: 0`, but `outcome: null`. Only the verify and repair paths record a job outcome, through the run verdict; the porcelain monitor loops (`--until-ready`, `--until-merged`, `--watch`) never call `markFinished`, so the finalizer reads a clean exit as `terminated`. The recipe's "0 green / 1 red" contract is therefore false for monitor jobs today. A follow-up PR records the outcome from those routes.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
